recording/src/nextcloud/talk/recording/BackendNotifier.py

```diff
@@ -59,16 +61,26 @@ def doRequest(backend, request, retries=3):
     :param backend: the backend to send the request to.
     :param request: the request to send.
     :param retries: the number of times to retry in case of failure.
+    :param library: the library to use for the request. Possible values are: 'urllib', 'requests'.
     """
-    context = None
-
-    if config.getBackendSkipVerify(backend):
-        context = ssl.create_default_context()
-        context.check_hostname = False
-        context.verify_mode = ssl.CERT_NONE
+    backendSkipVerify = config.getBackendSkipVerify(backend)
 
     try:
-        urlopen(request, context=context)
+        if isinstance(request, UrllibRequest):
```

Comment:
Rather than using both `Request` from `urllib` and `requests`, would it be possible to convert the `urllib` requests to use `requests` instead? I guess the `urllib` requests were kept just because it was the existing code (and in general I would prefer that :-) ), but in this case I think a unified approach would be better. Or is there any benefit in using `Request` from `urllib` in addition to the one from `requests`?

Reply:
From the docs, I get the impression that `requests` is more oriented towards `functional programming` and `urllib` towards `object-oriented programming`, although `object-oriented programming` is not impossible to achieve with `requests`. Kinda like PHP 4/5 vs PHP 7/8 differences.

Lately `requests` has been moving towards an `object-oriented` approach, so this difference is slowly becoming a non-issue.

If removing `urllib` entirely is not an issue, would it be acceptable to add a separate PR for such change, and keep the current implementation for this PR?

Reply:
> From the docs, I get the impression that `requests` is more oriented towards `functional programming` and `urllib` towards `object-oriented programming`, although `object-oriented programming` is not impossible to achieve with `requests`. Kinda like PHP 4/5 vs PHP 7/8 differences.
> 
> Lately `requests` has been moving towards an `object-oriented` approach, so this difference is slowly becoming a non-issue.

I am not familiar enough neither with `requests` nor `urllib`, so I can not provide any insight on that, sorry :-)

> If removing `urllib` entirely is not an issue, would it be acceptable to add a separate PR for such change, and keep the current implementation for this PR?

Yes, it would be fine. But I thought that it would just require changing `backendRequest = UrllibRequest(url, data, headers)` to `backendRequest = Request('POST', url, headers, data)` (based on the change done in `uploadRecording`) :thinking: Is something else required, or why would you like to do it in a different pull request? For the record, my idea was to first move the existing `urllib` code to `requests` and then, in a separate commit, adjust it to stream the upload.

Reply:
After reading all of the `urllib` calls, all of them are `POST` requests. Completely replacing `urllib` by `requests` should require minimal changes as you suggest.

Asked about a separate PR because of the description, but that can be adjusted anyways.

tl;dr if completely replacing `urllib` by `requests` is desired, so be it in this very PR (:

Reply:
> After reading all of the `urllib` calls, all of them are `POST` requests. Completely replacing `urllib` by `requests` should require minimal changes as you suggest.

He, now you made me wonder about what would happen if non-POST requests are needed at some point. But... let's worry about that in the future if needed ;-)

> Asked about a separate PR because of the description, but that can be adjusted anyways.

In general I prefer separate pull requests that are rebased one on top of the other (to nickvergessen's dismay :-P ), but in this case I think it is fine to do the groundwork of moving from `urllib` to `requests` as a separate commit in the same pull request.

> tl;dr if completely replacing `urllib` by `requests` is desired, so be it in this very PR (:

Perfect, thanks :-)